linetools/lists/linelist.py

```diff
@@ -593,48 +621,46 @@ def all_transitions(self, line):
-                return self.from_table_to_dict(tbl)
+                return lilu.from_table_to_dict(tbl)
 
         else:
             Z = None
-            for row in self._data:  # is this loop avoidable?
-                name = row['name']
-                # keep only the first part of name in linelist too
-                name = name.split(' ')[0]
-                if name == line:
-                    Z = row['Z']  # atomic number
-                    ie = row['ion']  # ionization estate
-                    Ej = row['Ej']  # Energy of lower level
-                    break
-            if Z is not None:
-                tbl = self.__getitem__((Z, ie))
-                # Make sure the lower energy level is the same too
-                #cond = tbl['Ej'] == Ej
-                cond = np.array([name1.split(' ')[0] == line for name1 in tbl['name']])
-                tbl = tbl[cond]
-                tbl.sort(['Ej','wrest'])
-                # For hydrogen/deuterium this contains deuterium/hydrogen;
-                # so let's get rid of them
-                #if (line == 'HI') or (line == 'DI'):
-                #    names = np.array(tbl['name'])
-                #    cond = np.array([l.startswith(line) for l in names])
-                #    tbl = tbl[cond]
-                if len(tbl) > 1:
-                    return tbl
-                else:  # this should be always len(tbl)==1 because Z is not None
-                    return self.from_table_to_dict(tbl)
-            else:
+            indices = np.where(self._data['ion_name'] == line)[0]
+            if len(indices) == 0:
                 raise ValueError(
                     'Line {} not found in the LineList: {}'.format(line, self.list))
+            else:
+                idx = indices[0]
+            Z = self._data['Z'][idx]  # atomic number
+            ie = self._data['ion'][idx]  # ionization state
+            Ej = self._data['Ej'][idx]  # Energy of lower level
+            tbl = self.__getitem__((Z, ie))
+            # Make sure the lower energy level is the same too
+            if debug:
+                pdb.set_trace()
+            cond = tbl['Ej'] == Ej
+            tbl = tbl[cond]
+            tbl.sort(['Ej','wrest'])
+            # For hydrogen/deuterium this contains deuterium/hydrogen;
+            # so let's get rid of them
+            if (line == 'HI') or (line == 'DI'):
+                names = np.array(tbl['name'])
+                cond = np.array([l.startswith(line) for l in names])
+                tbl = tbl[cond]
+            if len(tbl) > 1:
+                return tbl
+            else:  # this should be always len(tbl)==1 because Z is not None
+                return lilu.from_table_to_dict(tbl)
 
-    def strongest_transitions(self, line, wvlims, n_max=3, verbose=False):
-        """ Find the strongest transition for an ion
+    def strongest_transitions(self, line, wvlims, n_max=3, verbose=False,
```

Comment:
This method doesn't work either, also due to _data['ion_name'] KeyError.

Reply:
see comment below

Reply:
Works as expected now.